Tests

AsianHam opened the pull request.

AsianHam committed:
test setup + dotenv example + cors

created a data.ts file as a setup to more easily implement tests

added an example dotenv file

setup cors to handle requests originating from default react url

AsianHam committed:
Interface update

Updated the thumbnail property for the Recipe interface

Included an example image for a recipe

AsianHam committed:
ingredients update

updated ingredients to include a starred option

Jacksondr5 committed:
update yarn lock

AsianHam committed:
recipe format update

updated recipe interface and all correlated code

removed package-lock

AsianHam committed:
yup mongoose update

updated yup to allow a blank thumbnail

fixed updating recipes

Jacksondr5 commented on src/app.ts:
```diff
@@ -21,6 +22,11 @@ app.use(
 );
+app.use(
+  cors({
+    origin: "http://localhost:3000",
```
Make this an environment variable

Jacksondr5 commented on src/app.ts:
```diff
@@ -43,8 +49,9 @@ app.use(function errorHandler(
     });
   }
   if (err instanceof Error) {
+    console.log(Error);
```
Me thinks you dont want to log that

Jacksondr5 commented on src/recipe.ts:
```diff
@@ -1,14 +1,24 @@
+  show: boolean;
+}
+
+export interface Ingredients {
```
Singular please

Jacksondr5 commented on src/recipe.ts:
```diff
@@ -1,14 +1,24 @@
 
+export interface Thumbnail {
+  image: string;
+  show: boolean;
```
Isn't this UI only data?

AsianHam committed:
pr changes

changed cors url to an environment variable

removed ui specific properties from the interface
